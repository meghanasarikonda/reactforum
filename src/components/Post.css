.name {
  color: blueviolet
}

.comment{
  margin-bottom: 10px
}