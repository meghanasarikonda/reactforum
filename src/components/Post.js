import React from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

function User ({modal, toggle, comments, postbody}) {
  return(
    <Modal isOpen={modal} toggle={toggle}>
        <ModalHeader >Post details</ModalHeader>
          <ModalBody>
          {postbody()}
          </ModalBody>
          <ModalBody>
            <h1>break</h1>
            {comments.map(comment => 
              <div key={comment.id}>
                <h4>{comment.name}</h4>
                <p>{comment.body}</p>
                <h4>{comment.email}</h4>
              </div>
            )}
          </ModalBody>
          <ModalFooter>
            <Button color="primary" onClick={toggle}>Ok</Button>{' '}
            <Button color="secondary" onClick={toggle}>Cancel</Button>
          </ModalFooter>
        </Modal>
  )
}

export default User;
