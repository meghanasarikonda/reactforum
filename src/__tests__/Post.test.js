import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Post from '../components/Post';

function setup() {
  const props = {
    modal: jest.bool,
    toggle: jest.fn(),
    comments: [],
    postbody: jest.fn()
  }

  const enzymeWrapper = shallow(<Post {...props} />)

  return {
    props,
    enzymeWrapper
  }
}

describe('Post', () => {

  it('should render correctly', () => {
    const { enzymeWrapper } = setup()
    expect(shallowToJson(enzymeWrapper)).toMatchSnapshot();
  });

  it('should have four children', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.children().length).toBe(3);
  });

  it('shallow wrapper instance should be null', () => {
    const { enzymeWrapper } = setup()
    const instance = enzymeWrapper.instance();

    expect(instance).toBe(null);
  });

  it('should have modal', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.find('Modal').length).toBe(1);
  });

  it('should have modal header', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalHeader').length).toBe(1)
  });

  it('should have modal body', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalBody').length).toBe(1)
  });

  it('should have modal footer', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalFooter').length).toBe(1)
  });

  it('Modal Footer should have 2 buttons', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalFooter').children().find('Button').length).toBe(2)
  });

  it('Modal should have a true isOpen property', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({ modal: true })
    expect(enzymeWrapper.find('Modal').props().isOpen).toBe(true)
  });

  it('Modal should have a false isOpen property', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('Modal').props().isOpen).toBe(false)
  });

  it('should have primary and secondary buttons', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalFooter').find('Button').at(0).props().color).toBe('primary')
    expect(enzymeWrapper.find('ModalFooter').find('Button').at(1).props().color).toBe('secondary')
  })
 
})