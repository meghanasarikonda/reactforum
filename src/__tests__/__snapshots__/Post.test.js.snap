// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[MockFunction]}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    wrapTag="div"
  >
    Post details
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <div>
      <b />
      <p>
        by 
        <b />
      </p>
    </div>
    <p>
      Comments
    </p>
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      color="primary"
      onClick={[MockFunction]}
      tag="button"
    >
      Ok
    </Button>
     
    <Button
      color="secondary"
      onClick={[MockFunction]}
      tag="button"
    >
      Cancel
    </Button>
  </ModalFooter>
</Modal>
`;
